### PR TITLE
kubeadm: add flag to skip token print out

### PR DESCRIPTION
**Pull request by dmmcquay**
**What this PR does / why we need it**: When kubeadm init is used in an automated context, it still prints the token to standard out. When standard output ends up in a log file, it can be considered that the token is leaked there and can be compromised. This PR adds a flag you can select to not have it print out and explicitly disable this behavior.

This is a continuation from https://github.com/kubernetes/kubernetes/pull/42823 since it had to be closed.

**Which issue this PR fixes** : fixes #https://github.com/kubernetes/kubeadm/issues/160

**Special notes for your reviewer**: /cc @luxas @errordeveloper 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/types.go [hunk 1]
```diff
@@ -46,7 +46,8 @@ type MasterConfiguration struct {
 	// SelfHosted enables an alpha deployment type where the apiserver, scheduler, and
 	// controller manager are managed by Kubernetes itself. This option is likely to
 	// become the default in the future.
-	SelfHosted bool
+	SelfHosted     bool
+	SkipTokenPrint bool
```
Comment (luxas) [inline cmd/kubeadm/app/apis/kubeadm/types.go]: I still don't think this should be in the API.
It should be Init struct that's for CLI params

This is a CLI-specific thing, not something that should be a part of an API

**Comment by dmmcquay** [comment]
@luxas PTAL

**Comment by dmmcquay** [comment]
@luxas ping

File: cmd/kubeadm/app/cmd/init.go [hunk 2]
```diff
@@ -177,11 +182,12 @@ func NewInit(cfgPath string, cfg *kubeadmapi.MasterConfiguration, skipPreFlight
 	// Try to start the kubelet service in case it's inactive
 	preflight.TryStartKubelet()
 
-	return &Init{cfg: cfg}, nil
+	return &Init{cfg: cfg, skipTokenPrint: skipTokenPrint}, nil
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/init.go]: Idea for follow-up: I think we should kill NewInit() and move everything to Init.Run()

That way we wouldn't have to pass everything as args to this function
But can be handled later

**Review by luxas — APPROVED** [review]

**Comment by luxas** [comment]
Can you think of any way to cmd-test this? That would be great (and followed-up on)

/lgtm
/approve

**Comment by luxas** [comment]
Added the approved label manually

Requiring manual approval for hack/verify-flags (which nobody seems to know why it exists) is silly IMO (but probably just an oversight), we should probably let some dirs be changed without approval

cc @ixdy

**Comment by ixdy** [comment]
@luxas the issue you're looking for is https://github.com/kubernetes/test-infra/issues/1782 (and is poorly named)

**Comment by luxas** [comment]
cool, thanks @ixdy, subscribed myself to that issue

**Pull request merged**